lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -193,6 +213,23 @@ export class ClientCredentialClient extends BaseClient {
                 headers,
                 thumbprint
             );
+
+            // check if a new token is being requested because the cached token needs to be refreshed, not because it's expired
+            if (accessTokenCacheKey) {
+                // check if ESTS is down (status code 429 or 5xx)
+                if ((response.status === 429) || (response.status >= 500)) {
```

Comment:
Having thought about this some more, I think the safest is to throw on any exception and error code except 4xx